include/solvers/physics_assemblers.h

```diff
@@ -0,0 +1,71 @@
+// SPDX-FileCopyrightText: Copyright (c) 2021-2024 The Lethe Authors
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later
+
+#ifndef lethe_physics_assemblers_h
+#define lethe_physics_assemblers_h
+
+/**
+ * @brief Base class for assembler objects of physics and subequations.
+ *
+ * @tparam ScratchDataType Type of scratch data object used for linear system
+ * assembly.
+ *
+ * @tparam CopyDataType Type of copy data object used for transferring from
+ * local system assembly to global system.
+ */
+template <typename ScratchDataType, typename CopyDataType>
+class PhysicsAssemblerBase
+{
+public:
+  /**
+   * @brief Constructor of the base assembler for physics and subequations.
+   *
+   * @param[in] simulation_control The SimulationControl object that holds
+   * information related to the control of the steady-state or transient
+   * simulation.
+   */
+  PhysicsAssemblerBase(
+    const std::shared_ptr<SimulationControl> &simulation_control)
+    : simulation_control(simulation_control)
```

Comment:
I disagree with this choice. PhysicsAssemblerBase should not have the simulation control. Not all assemblers require a simulation control and I would want to keep the interface as light as possible. We will eventually have equations that are NOT transient...

Reply:
I'll remove it